### PR TITLE
Handle missing FSMap kwargs in FSStore

### LINKED ISSUE #775: Bug: Using GCS or S3, FSStore doesn't pass `check` and `create` kwargs to FSSPEC's get_mapper()
I think this is a bug, if everyone agrees I can fix it and do a PR.

#### Minimal, reproducible code sample, a copy-pastable example if possible

```python
import zarr

# Works fine with this as expected
url = '~/my_root.zarr'

# Doesn't work with:
url = r'gcs://my_bucket/my_root.zarr'
# or
url = r's3://my_bucket/my_root.zarr'

storage_options = {'key': my_key, 'secret': my_secret, 'token': my_token}  # Example for s3
store = zarr.storage.FSStore(url, check=True, create=True, **storage_options)
zarr.open(store, 'w')
```

#### Problem description

The above code fails because the FSMap created and used by FSStore did not receive the `check` and `create` kwargs (Traceback at the bottom of the post).

The `check` parameter is used to verify write access to remote stores.
The `create` function creates the root object/directory if it doesn't exist, so it can `check` for write access

The `fsspec.mapping.get_mapper` used in `zarr.storage.FSStore` expects explicit kwargs for `check` and `create` to send `FSMap` and passes `storage_options` aka. other `kwargs` to ONLY to `url_to_fs` ([here](https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/core.html?highlight=url_to_fs#)). The `FSMap` call here expects the `check` and `create` kwargs, which explicitly have to be passed to `get_mapper`.

The issue can be circumvented by dropping the write access checking, but that may not be always the case user wants.

The ideal fix is to make `zarr.storage.FSStore` to expect `check` and `create` kwargs; and default to False, just like `FSMap`.

#### Version and installation information

Please provide the following:

* ``zarr.__version__`` 2.8.1
* ``numcodecs.__version__`` 0.7.3
* Python 3.9.5
* Linux x86_64
* installed via conda

Traceback: 

```python
Traceback (most recent call last):
  File "/home/my_user/develop/my_library/benchmarks/test.py", line 5, in <module>
    store = zarr.storage.FSStore(url, check=True, create=True, **storage_options)
  File "/home/my_user/miniconda3/envs/my_library/lib/python3.9/site-packages/zarr/storage.py", line 1052, in __init__
    self.map = fsspec.get_mapper(url, **storage_options)
  File "/home/my_user/miniconda3/envs/my_library/lib/python3.9/site-packages/fsspec/mapping.py", line 224, in get_mapper
    return FSMap(root, fs, check, create, missing_exceptions=missing_exceptions)
  File "/home/my_user/miniconda3/envs/my_library/lib/python3.9/site-packages/fsspec/mapping.py", line 52, in __init__
    raise ValueError(
ValueError: Path my_bucket/my_root.zarr does not exist. Create  with the ``create=True`` keyword

```

### PR DESCRIPTION
**Pull request by tasansal**
This PR closes #775 

`fsspec.FSMap` created and used by `zarr.storage.FSStore` does not receive the `check` and `create` kwargs.
Fixed it by exposing `check` and `create` kwargs to `zarr.storage.FSStore`.
Defaults are False, just like `FSMap`.
Is this OK, or should we default them to `None` so it takes the default directly from `fsspec.FSMap` ?

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Comment by martindurant** [comment]
LGTM.

**Review by martindurant — APPROVED** [review]

**Review by rabernat — COMMENTED** [review]
Just a question about why this is needed.

File: zarr/storage.py [hunk 1]
```diff
@@ -1072,10 +1078,12 @@ def __init__(self, url, normalize_keys=False, key_separator=None,
                  mode='w',
                  exceptions=(KeyError, PermissionError, IOError),
                  dimension_separator=None,
+                 check=False,
+                 create=False,
```
[the remark this answers is not in the document]
Comment (tasansal) [reply zarr/storage.py:1082]: @rabernat

When check and/or create is passed, it errors out saying it got multiple values for it. One coming from the `**storage_options` in have in `FSStore` and one from `get_mapper`.

Maybe they changed the behavior, let me try it one more time.

Also, do I need to add this change to the release notes as a bug fix, or does someone else handle that before release?

[the remark this answers is not in the document]
Comment (tasansal) [reply zarr/storage.py:1082]: Sorry, correction. Looking at the `get_mapper` code again, 

All storage_options get passed to `url_to_fs` function: [link](https://github.com/intake/filesystem_spec/blob/06d1c7c9668377d37137e11dc7db84fee14d2d1b/fsspec/mapping.py#L228)

The `check` and `create` are actually in the `FSMap` call ([link](https://github.com/intake/filesystem_spec/blob/06d1c7c9668377d37137e11dc7db84fee14d2d1b/fsspec/mapping.py#L230))

**Comment by tasansal** [comment]
@rabernat @martindurant 

Digging deeper, I found some issues in `fsspec` + `s3fs` combination.

Details here. (Sounds like `fsspec` issue, so will create the main issue there)
https://github.com/dask/s3fs/issues/401#issuecomment-903896239

I think we can close this PR, and handle this in `fsspec` if you all agree.

**Comment by joshmoore** [comment]
> I think we can close this PR, and handle this in `fsspec` if you all agree.

Thanks, @tasansal. Closing but we can re-open if need be.

**Pull request closed without merging**